File: dataforge-core/src/main/java/hep/dataforge/io/markup/StringMarkupRenderer.java
package hep.dataforge.io.markup;

/**
 * A renderer that renders to string
 */
public class StringMarkupRenderer extends StreamMarkupRenderer {
    private StringBuilder builder = new StringBuilder();


    @Override
    protected synchronized void printText(String string) {
        builder.append(string);
    }

    @Override
    protected synchronized void ln() {
        builder.append("\n");
    }

    @Override
    public String toString() {
        return builder.toString();
    }

    public synchronized void reset(){
        this.builder = new StringBuilder();
    }
}

File: dataforge-markup/src/main/kotlin/hep/dataforge/markup/StringMarkupRenderer.kt
package hep.dataforge.markup.markup

/**
 * A renderer that renders to string
 */
class StringMarkupRenderer : StreamMarkupRenderer() {
    private var builder = StringBuilder()


    @Synchronized override fun printText(string: String) {
        builder.append(string)
    }

    @Synchronized override fun ln() {
        builder.append("\n")
    }

    override fun toString(): String {
        return builder.toString()
    }

    @Synchronized
    fun reset() {
        this.builder = StringBuilder()
    }
}
